spec.md

```diff
@@ -871,6 +878,14 @@ TODO: ord, chr.
 TODO: string.elems(), string.elem_ords(), string.codepoint_ords()
 ```
 
+### Unions
+
+Some functions may accept parameters of multiple types. In this
+case parameter is annotated with a union type `A|B`.
+
+User functions are limited to optional types, that is `None|A`,
```

Comment:
Nit: "user functions" might be ambiguous between application-defined vs starlark-programmer-defined. Maybe "Starlark-defined functions"? Not sure whether there's precedent terminology elsewhere in this spec.